pkg/source/control/reconciler/connection_pool.go

```diff
@@ -0,0 +1,232 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"knative.dev/pkg/logging"
+
+	"knative.dev/eventing-kafka/pkg/source/control/network"
+	ctrlservice "knative.dev/eventing-kafka/pkg/source/control/service"
+)
+
+const (
+	keepAlive = 30 * time.Second
+)
+
+type ControlPlaneConnectionPool struct {
```

Comment:
is there nothing we can (re)use for this?

Reply:
Not that I'm aware of :smile: